patches/0100-Add-OpenSSL-crypto-module.patch

```diff
@@ -90,10 +110,22 @@ index 0000000000..4de8d404f8
 +	"crypto/internal/boring/sig"
 +	"hash"
 +	"math/big"
++	"runtime"
++	"syscall"
 +)
 +
 +const Enabled = false
 +
++func init() {
++	if v, ok := syscall.Getenv("GOFIPS"); ok && v == "1" {
++		if runtime.GOOS != "linux" {
++			panic("FIPS mode requested (GOFIPS=1) but not supported on " + runtime.GOOS)
++		} else if !iscgo {
++			panic("FIPS mode requested (GOFIPS=1) but not supported with cgo disabled")
++		}
```

Comment:
Hmm, no `else` means if someone on Linux with cgo sets `GOFIPS=1` and `-tags gocrypt`, there's no panic, right?

This isn't necessarily bad (we could say gocrypt takes precedence), but it seems like the same type of scenario that should probably also panic to avoid false expectations.

(BTW, appreciate the detailed panic reasons in the messages here, I'm sure it'll help out with debugging enable issues later. 🙂)

Reply:
> Hmm, no else means if someone on Linux with cgo sets GOFIPS=1 and -tags gocrypt, there's no panic, right?

Yep. If a binary is built with `-tags gocrypt` it is because the developer intent is to always use Go crypto, regardless of the runtime configuration. If we panic when `GOFIPS=1` and the binary is built with `gocrypt`, then we would be forcing all those binaries to explicitly run with `GOFIPS=0` if `GOFIPS=1` system-wide, even for the once that do not care about FIPS, which would be really unfortunate.

Reply:
Ok, yeah, that makes sense.

Thinking about it a bit more, a big difference between cgo support and `-tags gocrypt` is that cgo is defined by your toolset, and a particular dev may not even know what toolset they're ending up using. But `-tags gocrypt` has to be very intentionally applied to the build--hard to imagine it being accidental.